core/util.c

```diff
@@ -217,3 +217,48 @@ void register_print_function(print_message_function_t* function, int log_level)
     print_message_function = function;
     print_message_level = log_level;
 }
+
+/**
+ * Increment the mixed radix number by one according to the permutation matrix.
+ */
+void mixed_radix_incr(mixed_radix_int_t* mixed) {
+	int i = 0;
+	while (i < mixed->size) {
+		int digit_to_increment = mixed->permutation[i];
```

Comment:
Also this seems dangerous. What if permutation holds an invalid index? It is used below unchecked.